### PR TITLE
Fix placement connection failures

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

> Note: Placement and its connection code in runtime do not have enough tests. This PR doesn't include the test to focus on the existing issues. Once this PR is merged, I will make the code testable and add unit-tests in the next PR.

This PR includes multiple fixes:

1. In self-host mode, actor cannot be discovered if multiple actor service dapr instances run because placement uses only ip address as a unique host.name -> Use `IP:Port` as host.name to distinguish the actor service instances which use the same ip address.
2. When the connection between dapr and placement is down, receiver loop and sender loop are trying to reconnect new grpc stream. It will establish duplicated stream channels. reconnection needs to happen in the single place to avoid this duplicated connections. 
3. Unnecessary send/retrieve retries -> use grpc.WithBlock() option and connection liveness flag to avoid the unnecessary retries.
4. In ReportDaprStatus(), using context.Done() may skip member removal step -> remove switch case and check the connection status only with stream.recv() error response to ensure that placement removes member from host list.
5. Add table version to the log
6. Add /healthz endpoint to actorload stateactor service. 

## Issue reference

#1732

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go
Comment (wcs1only) [inline pkg/actors/actors.go:450]: For the sake of not browning out an overloaded placement service, can we do exponential backoff on our retry interval?

File: pkg/actors/actors.go
Comment (wcs1only) [inline pkg/actors/actors.go:479]: Same as above, re: exponential backoff.

File: pkg/actors/actors.go
Comment (wcs1only) [inline pkg/actors/actors.go:497]: When you say CPU/Memory, do you mean realtime available CPU/Memory, or simply static host size? 

CPU/Memory tends to miss other performance problems such as disk and networking. Things like the ratio of busy time/idle time can capture system load more reliably. Or work queue depth, if your application has something like that.

File: pkg/actors/actors.go
Comment (wcs1only) [inline pkg/actors/actors.go:534]: Same comment, re: exponential backoff.

File: pkg/actors/actors.go
Comment (wcs1only) [inline pkg/actors/actors.go:544]: Same as above.

File: pkg/placement/placement.go
Comment (wcs1only) [inline pkg/placement/placement.go:231]: Not a review comment, but a question for my own learning. Does golang have a scoped lock? I know about defer to the end of the function execution, but what about to the end of the scope?

File: pkg/actors/actors.go
Comment (youngbupark) [reply pkg/actors/actors.go:497]: We will not support dynamic load balancing before 1.0. No plan but I would like to add comment, how we can use this number for the future. Disk/network can definitely be considered as load factor.

File: pkg/placement/placement.go
Comment (youngbupark) [reply pkg/placement/placement.go:231]: Personally, I do not prefer using `defer`. To implement the scoped lock, as you see the original code, local func and defer is one of the options.

```
func() {
   lock
  defer unlock
  ...
}
```

`func() {}` is the scope and defer will execute at the end of this scope.

File: pkg/actors/actors.go
Comment (youngbupark) [reply pkg/actors/actors.go:450]: for this case, linear retry is enough because this code inside this loop does not retry to establish the connection. 

We may need to add exponential backoff in `newPlacementStreamConn`

As I describe in the comment, this pr focuses on the current reported issue. Once this PR is merged, I will refactor to make it testable and add more tests.

File: pkg/actors/actors.go
Comment (youngbupark) [reply pkg/actors/actors.go:479]: Same above.

File: pkg/actors/actors.go
Comment (youngbupark) [reply pkg/actors/actors.go:544]: I described above.

**Comment by youngbupark** [comment]
I plan to refactor the code and add unit-test later. but I have to do it now :) @yaron2 This focuses on the issue fix. but I will create new PR to make the code testable and add more test.

File: pkg/actors/actors.go
Comment (wcs1only) [reply pkg/actors/actors.go:497]: Well, specifically in an event driven system, a good measure that can capture slowdowns of any kind is (time_processing_events)/(time_processing_events + time_waiting_for_new_events). This manages to capture the aggregate of CPU + Network + disk + memory + any other thing that might slow down event processing.

**Comment by youngbupark** [comment]
> I plan to refactor the code and add unit-test later. but I have to do it now because of codecov :) @yaron2 This focuses on the issue fix. but I will create new PR to make the code testable and add more test.

@yaron2 @wcs1only / I tried to refactor the code with the unit-test, but I may not have a time to finish and validate it again. this change has been validated by @KaiWalter

https://github.com/dapr/dapr/issues/1732#issuecomment-697434814

in 0.12.0 milestone, I will bring the better fix to address the problems in our load test.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**
